src/backend/dx12/src/lib.rs

```diff
@@ -1172,3 +1090,116 @@ fn validate_line_width(width: f32) {
     // Simply assert and no-op because DX12 never exposes `Features::LINE_WIDTH`
     assert_eq!(width, 1.0);
 }
+
+pub struct FormatProperties(
+    [Mutex<Option<f::Properties>>; f::NUM_FORMATS],
+    native::Device,
+);
+
+impl Drop for FormatProperties {
+    fn drop(&mut self) {
+        unsafe {
+            self.1.destroy();
+        }
+    }
+}
+
+impl FormatProperties {
+    fn new(device: native::Device) -> Self {
+        let mut arr: [Mutex<Option<f::Properties>>; f::NUM_FORMATS] =
```

Comment:
let's just make it a `Box<[]>`, given that we can construct a `Vec` and convert it into that with https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.into_boxed_slice